### ISSUE TITLE
[feature request] command line app

### ISSUE THREAD

**Issue by fabrizziosoares** [gechr/WhichSpace#25]
is it possible to make is a command line app to print the current space?

**Comment by vemv** [comment]
+1, I'd love this and am willing to sponsor this feature

**Comment by davidpurnell** [comment]
Hello,

I had a need for a command line utility as well so I grabbed the WhichSpace function that queries the spaces info and wrapped it in a command line swift app.

You can find it here: https://github.com/davidpurnell/SpaceInfo

**Comment by vemv** [comment]
Cheers! Starred, will give it a spin

I discovered / was using https://github.com/alt-jero/McSpaces in the meantime, it works fine, but bit of a raw one.

**Comment by davidpurnell** [comment]
Ah, McSpaces is giving UUIDs.  My version outputs integers incremented from 1.  I'm using it in some Keyboard Maestro macros so my use case required simple numbering of spaces. Hope that works for you!

**Comment by vemv** [comment]
Yes actually my usage of McSpaces was pretty hacky. It gives you a blank UUID for Space #1, and an actual UUID for the rest.

By coincidence, I was only interested in querying "am I in space #1?" so getting a blank UUID was enough 😄

A number-based approach is more robust and allows more use cases.

**Comment by jim-sauer** [comment]
> I have several KM macros that move windows around between multiple displays and spaces. These macros need to know what current space and/or display index the frontmost window is located on.

@davidpurnell, SpaceInfo looks cool. You may know this already, but if WhichSpace is running, you can get the current space with AppleScript:

```applescript
tell application "System Events" to tell process "WhichSpace" to set temp to (title of menu bar items of menu bar 1)
return item 1 of temp
```

I'm going to check out your macros on GitHub. Have you shared them on the [KM Forum](https://forum.keyboardmaestro.com/latest)? If not consider doing so. It's a great place to share KM ideas.

I've also developed a set of KM macros to manage spaces; there *are* on the forum: [Desktop Spaces • Macros for Navigation and Window Management](https://forum.keyboardmaestro.com/t/macros-desktop-spaces-macros-for-navigation-and-window-management-v1-1/21507/6)

**Comment by davidpurnell** [comment]
Yep. I used to query Whichspace this way to get the current space index and then used that in earlier versions of my KM macros.  Pretty sure I've seen some of your macros on the forum. I'm a lurker there but perhaps it's time to start posting!
